llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -346,7 +346,9 @@ void RuntimePointerChecking::tryToCreateDiffCheck(
     auto *SinkStartAR = cast<SCEVAddRecExpr>(SinkStartInt);
     const Loop *StartARLoop = SrcStartAR->getLoop();
     if (StartARLoop == SinkStartAR->getLoop() &&
-        StartARLoop == InnerLoop->getParentLoop()) {
+        StartARLoop == InnerLoop->getParentLoop() &&
+        SrcStartAR->getStepRecurrence(*SE) !=
```

Comment:
Would be good to add a comment here explaining the check

Reply:
Hi @fhahn, sorry I only just saw this now! I find with github it's really easy to miss review comments compared to Phabricator...

Reply:
Yeah the UI does unfortunately sometimes hide some comments :(